magma/backend/coreir/insert_wrap_casts.py

```diff
@@ -27,15 +27,20 @@ def simulate(self, value_store, state_store):
                 value_store.set_value(self.out, input_val)
         return Wrap
 
+    def _recurse(self, port, definition):
+        wrapped = False
```

Comment:
`return any(self.wrap_if_named_type(t, definition) for t in port)`

Reply:
We actually don't want it to short circuit because we want it to wrap all children that have named types. The wrapped bool logic was for early exiting in arrays (since if one child wasn't wrapped we didn't need to traverse the rest)

Reply:
we could do

```python
return any([self.wrap_if_named_type(t, definition) for t in port])
```

or even

```
# NOTE(leonardt): We can't short circuit, since wrap_if_named_type needs to be called on all elements, so we materialize them into a list rather than a generator.
t_wrapped = [self.wrap_if_named_type(t, definition) for t in port]
return any(t_wrapped)
```

might be more readable but nbd either way, leaving it up to you